android/src/main/java/com/zoontek/rnpermissions/RNPermissionsModuleImpl.kt

```diff
@@ -0,0 +1,241 @@
+      val permission = permissions.getString(i)
+
+      output.putBoolean(
+        permission,
+        context.checkSelfPermission(permission) == PackageManager.PERMISSION_GRANTED
+      )
+    }
+
+    promise.resolve(output)
+  }
+
+  fun request(
+    reactContext: ReactApplicationContext,
+    listener: PermissionListener,
+    callbacks: SparseArray<Callback>,
+    permission: String,
+    promise: Promise
+  ) {
+    val context = reactContext.baseContext
+
+    if (context.checkSelfPermission(permission) == PackageManager.PERMISSION_GRANTED) {
+      return promise.resolve(GRANTED)
+    }
+
+    try {
+      val activity = getPermissionAwareActivity(reactContext)
+
+      callbacks.put(
+        requestCode,
+        Callback { args ->
+          val results = args[0] as IntArray
+          val callbackActivity = args[1] as PermissionAwareActivity
+
+          val status = when {
+            results.isNotEmpty() && results[0] == PackageManager.PERMISSION_GRANTED -> GRANTED
+            callbackActivity.shouldShowRequestPermissionRationale(permission) -> DENIED
+            else -> BLOCKED
+          }
+
+          promise.resolve(status)
+        })
+
+      activity.requestPermissions(arrayOf(permission), requestCode, listener)
+      requestCode++
+    } catch (e: IllegalStateException) {
+      promise.reject(ERROR_INVALID_ACTIVITY, e)
+    }
+  }
+
+  // Only used on Android < 13 (the POST_NOTIFICATIONS runtime permission isn't available)
+  fun requestNotifications(reactContext: ReactApplicationContext, promise: Promise) {
+    val granted = NotificationManagerCompat.from(reactContext).areNotificationsEnabled()
+
+    val output = Arguments.createMap().apply {
+      putString("status", if (granted) GRANTED else BLOCKED)
+      putMap("settings", Arguments.createMap())
+    }
+
+    promise.resolve(output)
+  }
+
+  fun requestMultiple(
+    reactContext: ReactApplicationContext,
+    listener: PermissionListener,
+    callbacks: SparseArray<Callback>,
+    permissions: ReadableArray,
+    promise: Promise
+  ) {
+    val grantedPermissions: WritableMap = WritableNativeMap()
+    val permissionsToCheck = ArrayList<String>()
+    var checkedPermissionsCount = 0
+    val context = reactContext.baseContext
+
+    for (i in 0 until permissions.size()) {
+      val permission = permissions.getString(i)
+
+      if (context.checkSelfPermission(permission) == PackageManager.PERMISSION_GRANTED) {
+        grantedPermissions.putString(permission, GRANTED)
+        checkedPermissionsCount++
+      } else {
+        permissionsToCheck.add(permission)
+      }
+    }
+
+    if (permissions.size() == checkedPermissionsCount) {
+      return promise.resolve(grantedPermissions)
+    }
+
+    try {
+      val activity = getPermissionAwareActivity(reactContext)
+
+      callbacks.put(
+        requestCode,
+        Callback { args ->
+          val results = args[0] as IntArray
+          val callbackActivity = args[1] as PermissionAwareActivity
```

Comment:
Is this a safe operation?

Reply:
Yes, it was the same in the  Java codebase / in react native core